linker/descriptors.go

```diff
@@ -1229,29 +1282,41 @@ func (f *fldDescriptor) IsWeak() bool {
 }
 
 func (f *fldDescriptor) IsPacked() bool {
-	opts := f.proto.GetOptions()
-	if opts.GetPacked() {
-		return true
-	}
-	if opts != nil && opts.Packed != nil {
-		// explicitly not packed
-		return false
-	}
+	switch f.Syntax() {
+	case protoreflect.Proto2:
+		return f.proto.GetOptions().GetPacked()
+
+	case protoreflect.Proto3:
+		opts := f.proto.GetOptions()
+		if opts.GetPacked() {
+			// explicitly packed
+			return true
+		}
+		if opts != nil && opts.Packed != nil {
+			// explicitly not packed
+			return false
+		}
+		// proto3 defaults to packed for repeated scalar numeric fields
+		if f.proto.GetLabel() != descriptorpb.FieldDescriptorProto_LABEL_REPEATED {
+			return false
+		}
+		switch f.proto.GetType() {
+		case descriptorpb.FieldDescriptorProto_TYPE_GROUP, descriptorpb.FieldDescriptorProto_TYPE_MESSAGE,
+			descriptorpb.FieldDescriptorProto_TYPE_BYTES, descriptorpb.FieldDescriptorProto_TYPE_STRING:
+			return false
+		default:
+			// all others can be packed
+			return true
+		}
+
+	case protoreflect.Editions:
+		fieldEncoding := resolveFeature(f, func(features *descriptorpb.FeatureSet) (descriptorpb.FeatureSet_RepeatedFieldEncoding, bool) {
+			return features.GetRepeatedFieldEncoding(), features != nil && features.RepeatedFieldEncoding != nil
+		})
+		return fieldEncoding == descriptorpb.FeatureSet_PACKED
```

Comment:
The proto2 / proto3  option `packed=true` cannot be set on`bytes`, `string`, etc. But the resolved edition feature can be `PACKED` for these types. 

So it's very possible that `IsPacked` returns `true` for a field that cannot be packed (unless the wire format changes drastically).

What do you think about returning `false` in this case, similar to the case above for proto3?

Reply:
Ah, right. The compiler prevents the packed setting from being set that way on those fields, but you're right that it could be inherited from a file option, and then report the wrong thing here. Good catch!

Reply:
Done!